### PR TITLE
directory-menu@torchipeppo: Add execution to 'Open' functions

### PR DESCRIPTION
**Pull request by rcalixte**
Add translation template
Add a terminal option for non-Debian-based distros

This addresses the execution functionality of the applet.

cc @torchipeppo

### CODE REVIEW

File: directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py [hunk 1]
```diff
@@ -136,12 +137,17 @@ def launch(self, uri, timestamp):
             "uri": uri,
             "timestamp": timestamp,
         }))
+        subprocess.run(['/usr/bin/xdg-open', uri], check=False)
```
Comment (torchipeppo) [inline directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py:140]: [1/3] In my design for this version, the launch and open_terminal_at_path functions *are* supposed to do nothing but printing. This sends the JSON string on stdout to be received by applet.js, which then takes care of executing the requested action.

File: directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py [hunk 2]
```diff
@@ -136,12 +137,17 @@ def launch(self, uri, timestamp):
...
 
     def open_terminal_at_path(self, path):
         print(json.dumps({
             "action": "open_terminal_at_path",
             "path": path,
         }))
+        terminal = self.terminal_cmd if self.terminal_cmd else "/usr/bin/x-terminal-emulator --working-directory"
```
Comment (torchipeppo) [inline directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py:147]: [2/3] Same as the comment for the launch function above

File: directory-menu@torchipeppo/files/directory-menu@torchipeppo/applet.js [hunk 3]
```diff
@@ -163,6 +164,7 @@ class CassettoneApplet extends Applet.IconApplet {
             "x": x,
             "y": y,
             "orientation": o,
+            "terminal_cmd": this.terminal_cmd,
```
Comment (torchipeppo) [inline directory-menu@torchipeppo/files/directory-menu@torchipeppo/applet.js:167]: [3/3] The callback function  beginning at line 170 or 172 is the one that actually takes care of executing the action clicked on the menu, as per the comments on the Python file. The callback is executed after the Python process terminates, with the "response" argument being whatever the process wrote to stdout.

**Review by torchipeppo — COMMENTED** [review]
Hi, I notice that this PR is adding execution functionality to the Python code which is already taken care of on the GJS side, was there some problem with the submitted version?

I added line-specific comments to illustrate the communication between the Python and GJS sides.

I will admit this is not the most elegant design, but it is the best I could do with my knowledge: I was not aware of `xdg-open`, and `Gio.AppInfo.launch_default_for_uri_async` and `GLib.spawn_async` gave me problems I can't remember right now in Python.

File: directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py (hunk 1 above)
Comment (rcalixte) [reply directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py:140]: Hmm. That wasn't working for me. I'll test this on another machine. I do like the gsettings lookup as a cleaner alternative that doesn't require a new preference.

Comment (rcalixte) [reply directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py:140]: Did you test this with other terminals than gnome-terminal?

Comment (torchipeppo) [reply directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py:140]: My default terminal is `terminator` actually

Comment (rcalixte) [reply directory-menu@torchipeppo/files/directory-menu@torchipeppo/popup_menu.py:140]: > My default terminal is `terminator` actually

Same! I'm a newcomer as of a few months. You have great taste! 😁

**Comment by rcalixte** [comment]
@torchipeppo This is ready for your review. 🙏🏼

**Comment by torchipeppo** [comment]
The code looks good, and from a quick test it looks like this version can open folders, terminals and files on my Mint 20.2 machine without issue. Thank you so much for your help :D

Just a couple minor points, to align the linuxmint repo with my personal one with all the dev history:
* I'd like to move your code there, with your permission;
* And I think we should increment the version number, since v0.3 is supposed to be my submission from a few hours ago. Moving the execution logic sounds like more than patch-level to me, so maybe this can be v0.4.0?

**Comment by torchipeppo** [comment]
Edited the previous message with links.

**Review by torchipeppo — CHANGES_REQUESTED** [review]
Remembered where the review button was

**Comment by rcalixte** [comment]
> * I'd like to put your code there too, with your permission;

No permission needed! Feel free! That's why we open source. 🤓



> * Moving the execution logic sounds like more than patch-level to me, so maybe this can be v0.4.0?

You got it.

**Comment by rcalixte** [comment]
Alright. Now we're ready. 😄 


Feel free to update the README when you're ready too. We'll get that merged for you quickly.

**Comment by torchipeppo** [comment]
And with that, I'd say we can merge this, as far as I'm concerned! ^_^

**Comment by torchipeppo** [comment]
I don't think I can push anything on this PR, though, right?

**Review by torchipeppo — APPROVED** [review]
Only the version number changed since the last review, so it's good

**Comment by rcalixte** [comment]
> I don't think I can push anything on this PR, though, right? So this should be merged first, and then I'd open a PR for the readme

Correct. Sorry I didn't make that part clear.

**Pull request merged**